main.cpp

```diff
@@ -1,25 +1,57 @@
 #include <iostream>
+#include <getopt.h>
 #include "common.h"
-#include "md5-duplicate-detection.h"
 #include "encrypt-decrypt.h"
 #include "files.h"
 
-int main() {
-    std::cout << "Hello, World!" << std::endl;
-    /*
-    std::string directory = "some directory";
+int main(int argc, char *argv[]) {
+
+    int opt;
+    const char *directory;
     std::vector<fs::path> files;
-    getFiles(directory, files);
-    for (fs::path i: files)
-        std::cout << i.string() << ' ';
-        */
 
-//    fs::path A("test-files/A");
-//    fs::path B("test-files/B");
-//    is_data_duplicate(A,B);
+    while (true) { //getopt creates a data-structure that can be looped through from the argv array to process the parameters and their arguments
+
+        int option_index = 0;
+        static struct option long_options[] =
+                {
+                        {"help",     no_argument, NULL,  'h' },
+                        {"encrypt",  no_argument, NULL,  'e' },
+                        {"decrypt",  no_argument, NULL,  'd' },
+                        {NULL,      0,    NULL,   0  }
+                };
+        opt = getopt_long(argc, argv, "he:d:", long_options, &option_index);
```

Comment:
You should parse argv entirely before performing any sub-routines. Thereby allowing you to determine if the command makes sense. Which will allow you to tell the user what they've done wrong. I would even recommend having the options be mutually exclusive, if this getopts library supports that or if you can determine a way to enforce it.